### DISCUSSION TITLE
Visualizing large scale cross-linking data using circos plots

### DISCUSSION THREAD

**Discussion by alessiodiianni** [moshi4/pyCirclize#24] [General]
Hi, 
I just started using pyCirclize and the whole package is amazing! I am currently trying to use it to map cross-linking mass spectrometry data (in terms of circos links) between proteins. I managed to do so, but only from manual definition of sectors and links between sectors and I would like ideally to do it from excel/csv files imported as pandas dataframes.
Could you help me on that?

**Comment by moshi4** [comment]
Hi @alessiodiianni,

I have never handled mass spectrometry data. I also do not know what kind of values you expect to contain in your input data and what kind of output diagram you want. Therefore, it is difficult for me to answer anything that will help you.

If you can clarify the inputs and outputs you are assuming, I may be able to give you some hints.

**Comment by alessiodiianni** [comment]
![image](https://github.com/moshi4/pyCirclize/assets/127302106/70e9b65d-9af8-4197-9e5f-a215b2be2f82)
This is a screenshot of the code I was testing with five sectors as example. It just requires very easy outer tracks with the protein sequences and writing circos links for the aminoacid positions found to be cross-linked. I saw in your repository you can load genomic dataset or create matrixes from pandas dataframes and I was wondering if I could to something similar so to automate large scale cross-links mapping. For this, I was thinking to have a .csv/excel file in which I have in the first sheet the proteins found to be crosslinked and their aminoacid sequence length (instead of chromosome kb) and in the second one a table in which I have protein 1 site 1 protein 2 site 2 so that you can directly trace links connections from there.

**Reply by moshi4** [reply]
> I was wondering if I could to something similar so to automate large scale cross-links mapping

pyCirclize does not have the functionality to automate this kind of work. Only Chord Diagram can plot from matrix data.

Looking at the example code you shown, it does not seem that difficult to implement `cross-links mapping` you mention by yourself.

**Reply by alessiodiianni** [reply]
Ok I hoped there was already something which you could use to define single site interactions by giving predefined tables with the required info. Thanks for the answer, I'll try it out.
